core/src/com/unciv/ui/components/ISortableGridContentProvider.kt

```diff
@@ -0,0 +1,65 @@
+package com.unciv.ui.components
+
+import com.badlogic.gdx.scenes.scene2d.Actor
+import com.unciv.logic.GameInfo
+
+/**
+ * This defines all behaviour of a sortable Grid per column through overridable parts:
+ * - [isVisible] can hide a column
+ * - [align], [fillX], [expandX], [equalizeHeight] control geometry
+ * - [getComparator] or [getEntryValue] control sorting, [defaultDescending] the initial order
+ * - [headerTip] and [getHeaderIcon] define how the header row looks
+ * - [getEntryValue] or [getEntryActor] define what the cells display
+ * - [getEntryValue] or [getTotalsActor] define what the totals row displays
+ * @param IT The item type - what defines the row
+ * @param ST The parent Screen Type - passed to [getEntryActor] for potential OnClick calls
+ */
+interface ISortableGridContentProvider<IT, ST> {
```

Comment:
If this is supposed to be a screen type then we should harden the interface to only allow screen types